### PR TITLE
Animal cooking oil and baking fixes

### PR DESCRIPTION
**Pull request by anonym**
#### Summary
SUMMARY: Content "Add animal cooking oil, add recipe for canning beans, and rebalance various foods and recipes"

#### Purpose of change
I actually got pulled into this rabbit hole of fixing baking-related inconsistencies when I experienced #29076 (but @KorGgenT was faster than me!). And that is what the purpose is, fixing a few bugs, (and a typo that made chocolate pancakes 10x more nutritious than they should be) and filling a few wholes (I can't make edible cannabis without chocolate?).

#### Describe alternatives you've considered
Instead of adding animal cooking oil I suppose I could have made lard/tallow more granular by making each charge an 1/8th of was they are now.

#### Additional context
Perhaps you, @KorGgenT, want to have a look since this is an area you've been working on lately!

### CODE REVIEW

**Comment by KorGgenT** [comment]
note: if the travis test fails, it could possibly be because of the recipes. i'll check back in a few hours when it finishes

**Comment by KorGgenT** [comment]
also: you only want one summary line. you've got four, so you want to amalgamate those summaries

**Comment by KorGgenT** [comment]
image
(you can ignore the fish soup because i'm fixing it in another PR)
if the "average" is higher than the upper bound, then that means the sum of the calories of the ingredients is too high. you can either fix that by increasing the number of charges the recipe makes, or reducing the overall calorie count. semi-recently i put a test into place that checks the ingredients of a recipe against its output to make sure that they're pretty close to the same calories input and output.

**Comment by anonym** [comment]
Cheers, @KorGgenT! Gonna start running `cata_test` before each PR from now on. It passes for me now, except `soup_fish_jarred_3l`, and also `soup_meat_jarred_3l` but I'm guessing it is unrelated (?).

I ended up also fixing stuff related to beans/rice, and added recipes for canning beans.

**Comment by KorGgenT** [comment]
try rebasing to master. it might be related.

File: data/json/recipes/recipe_food.json [hunk 1]
```diff
@@ -2544,7 +2589,7 @@
     "batch_time_factors": [ 82, 4 ],
     "qualities": [ { "id": "COOK", "level": 2 } ],
     "tools": [ [ [ "surface_heat", 2, "LIST" ] ] ],
-    "components": [ [ [ "dry_beans", 1 ], [ "raw_beans", 1 ] ], [ [ "water", 1 ], [ "water_clean", 1 ] ] ]
+    "components": [ [ [ "dry_beans", 1 ], [ "raw_beans", 1 ], [ "can_beans", 1 ] ], [ [ "water", 1 ], [ "water_clean", 1 ] ] ]
```
Comment (ZhilkinSerg) [inline data/json/recipes/recipe_food.json]:
```suggestion
    "components": [
      [ [ "can_beans", 1 ], [ "dry_beans", 1 ], [ "raw_beans", 1 ] ],
      [ [ "dry_rice", 1 ] ],
      [ [ "water", 1 ], [ "water_clean", 1 ] ]
    ]
```

**Comment by ZhilkinSerg** [comment]
image

**Comment by anonym** [comment]
So `git bisect` lead me to the reason those soups failed in the test suite: the nerf of cooked beans. Looking deeper I could see that the real problems was that `soup_meat_jarred_3l` required twice the amounts of components it should, and soup_fish was basically worth half the calories it should, both which I fixed. Now the tests pass for me locally!

However, in the soup recipes the "filler" ingredient is extremely unbalanced, for example a unit of 15 kcal celery is equivalent to a unit of 222 kcal rice. But I must stop delving deeper into this rabbit hole! :)

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/cooked-beans-recipe-beans-and-rice-recipe/19512/3

**Comment by anonym** [comment]
@ZhilkinSerg, see the mention on discord above. It seems the commit we co-authored (1bccf0554c1eca3f3233f1e78c8393a74b2b5c3c) is wrong, and adds rice as a requirement for cooked beans. Can you please revert it?

**Comment by ZhilkinSerg** [comment]
Just create a PR to fix any issues found.

**Pull request merged**